agg-ast/schema_derivation/src/negative_normalize.rs

```diff
@@ -228,30 +260,108 @@ impl NegativeNormalize<Expression> for Expression {
+                    | UntaggedOperatorName::SQLLog | UntaggedOperatorName::SQLMod
+                    | UntaggedOperatorName::SQLNeg | UntaggedOperatorName::SQLPos
+                    | UntaggedOperatorName::SQLRound | UntaggedOperatorName::SQLSize
+                    | UntaggedOperatorName::SQLSqrt | UntaggedOperatorName::BitAnd
+                    | UntaggedOperatorName::BitNot | UntaggedOperatorName::BitOr
+                    | UntaggedOperatorName::BitXor | UntaggedOperatorName::BsonSize
+                    | UntaggedOperatorName::CovariancePop | UntaggedOperatorName::CovarianceSamp
+                    | UntaggedOperatorName::StdDevPop| UntaggedOperatorName::StdDevSamp
+                    | UntaggedOperatorName::Round | UntaggedOperatorName::ToDecimal
+                    | UntaggedOperatorName::ToDouble | UntaggedOperatorName::ToLong
+                    | UntaggedOperatorName::NumberDouble | UntaggedOperatorName::SQLSum
+                    | UntaggedOperatorName::SQLTan => {
                         let null_check = wrap_in_null_or_missing_check!(self.clone());
                         let zero_check = wrap_in_zero_check!(self.clone());
                         (UntaggedOperatorName::Or, vec![null_check, zero_check])
                     }
+                    // The following operators may evaluate to 0 or false
+                    UntaggedOperatorName::Is | UntaggedOperatorName::Locf => {
+                        let zero_check = wrap_in_zero_check!(self.clone());
+                        let false_check = wrap_in_false_check!(self.clone());
+                        (UntaggedOperatorName::Or, vec![zero_check, false_check])
+                    }
                     // The following operators may evaluate to the falsy values missing, null, 0, or
                     // false, so the negation asserts equality to any of those values.
-                    UntaggedOperatorName::First | UntaggedOperatorName::IfNull | UntaggedOperatorName::Last | UntaggedOperatorName::Literal | UntaggedOperatorName::Max | UntaggedOperatorName::Min => {
+                    UntaggedOperatorName::Coalesce | UntaggedOperatorName::NullIf
+                    | UntaggedOperatorName::First | UntaggedOperatorName::IfNull
+                    | UntaggedOperatorName::Last | UntaggedOperatorName::Literal
+                    | UntaggedOperatorName::Max | UntaggedOperatorName::Min => {
                         let null_check = wrap_in_null_or_missing_check!(self.clone());
                         let zero_check = wrap_in_zero_check!(self.clone());
                         let false_check = wrap_in_false_check!(self.clone());
                         (UntaggedOperatorName::Or, vec![null_check, zero_check, false_check])
                     }
+                    // the following can only evaluate to true or false
+                    | UntaggedOperatorName::MQLBetween => {
+                        let false_check = wrap_in_false_check!(self.clone());
```

Comment:
nit but this and the following null check only can probably just happen on one line, that's how we do it in TaggedOperator for types that only have one check